.pipelines/templates/template-prod-e2e-steps.yml

```diff
@@ -11,6 +11,14 @@ steps:
     export LOCATION=${{ parameters.location }}
     export AZURE_SUBSCRIPTION_ID=${{ parameters.subscription }}
 
+    ifeq (${AZURE_ENVIRONMENT},"AzureUSGovernmentCloud")
```

Comment:
I'm not sure I'm mentally ready to merge the .pipelines or Makefile changes in this PR.  You can try to convince me though.

My concerns are:
1) we're in the middle of trying to simplify pipelines, but this adds complexity which isn't actually in use yet
2) we don't yet own the ACR and storage account names in govcloud, if someone else has them this logic would have  to change
3) I would like to have a process to promote built images from int->prod but we don't yet know how we'll want to architect that for govt cloud.  There's no guarantee that the changes proposed here will take us in the right direction.

We'll need to do these things, I just think it's a bit early still.

Reply:
OK, sounds good. I'll pull these commits out.